synapse/rest/client/v2_alpha/register.py

```diff
@@ -497,6 +496,15 @@ async def on_POST(self, request):
             registered_user_id = await self.auth_handler.get_session_data(
                 session_id, "registered_user_id", None
             )
+            # If a password hash was previously stored we will not attempt to
+            # re-hash and store it.
+            #
+            # Note that if the password changes throughout the authentication
+            # flow this might break, but the data is meant to be consistent
+            # throughout the flow.
+            password_hash = await self.auth_handler.get_session_data(
+                session_id, "password_hash", None
+            )
```

Comment:
I struggled for a while to understand what was going on here, and tbh the comment didn't do much to help.

I think this comment - especially the bit about what happens if the password changes mid-flow - should move down to inside the `except InteractiveAuthIncompleteError` block at line 533ish, and we can just say "extract the previously-hashed password from the session" here.

Reply:
I struggled with where to put this, moving it further up sounds like it would be an improvement.

Is there a particular piece of knowledge that should be documented in comments?

Reply:
> I struggled with where to put this, moving it further up sounds like it would be an improvement.

I'm mostly suggesting the code gets left where it is, and the comment moves *down*.

> Is there a particular piece of knowledge that should be documented in comments?

I don't think so, tbh. I think this code speaks for itself, but I only realised that after I stopped trying to understand the comment, which doesn't really describe what is happening here :)

Reply:
I re-arranged and re-worded the comments in e05118bfccfb95c4e0ffe5f5f08200d4ee9e88fb, I'm hoping that helps?

Reply:
yup, lgtm.